File: sach/de3399.py
//DE3399.PY - Tran Huu Nam - hsgtin.vn - 2024
import sys
from typing import List, Tuple
def main():
    input = sys.stdin.read
    data = input().split()
    index = 0
    n = int(data[index])
    m = int(data[index + 1])
    index += 2
    a: List[Tuple[int, int]] = []
    c: List[int] = []
    for _ in range(n):
        a.append((int(data[index]), int(data[index + 1])))
        index += 2
    for _ in range(m):
        c.append(int(data[index]))
        index += 1
    c.sort(reverse=True)
    a.sort(key=lambda x: (-x[1], -x[0]))
    j = 0
    for i in range(n):
        if a[i][0] <= c[j]:
            j += 1
        if j == m:
            break
    print(j)
if __name__ == "__main__":
    main()

File: sach/de3399.cpp
//DE3399.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
#define f first
#define s second
#define all(x) x.begin(), x.end()
typedef long long ll;
typedef pair <int, int> ii;
const int N = 2e5 + 1;
int main() {
    ios_base::sync_with_stdio(0); cin.tie(0);
    int n, m; cin >> n >> m;
    vector <ii> a(n);      vector <int> c(m);
    for(auto &x: a) cin >> x.f >> x.s;
    for(auto &x: c) cin >> x;
    sort(all(c), greater <int> ());
    sort(all(a), [] (ii x, ii y) {
        if (x.s != y.s) return x.s > y.s;
        return x.f > y.f;
    });
    int j = 0;
    for(int i = 0; i < n && j < m; i++) {
        if (a[i].f <= c[j]) j++;
    }
    cout << j << "\n";
}

